test/utils/decorators.py

```diff
@@ -10,289 +10,124 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Decorators for using with IBM Provider unit tests.
-
-    Environment variables used by the decorators:
-        * QISKIT_IBM_API_TOKEN: default API token to use.
-        * QISKIT_IBM_API_URL: default API url to use.
-        * QISKIT_IBM_HGP: default hub/group/project to use.
-        * QISKIT_IBM_PRIVATE_HGP: hub/group/project to use for private jobs.
-        * QISKIT_IBM_DEVICE: default device to use.
-        * QISKIT_IBM_USE_STAGING_CREDENTIALS: True if use staging credentials.
-        * QISKIT_IBM_STAGING_API_TOKEN: staging API token to use.
-        * QISKIT_IBM_STAGING_API_URL: staging API url to use.
-        * QISKIT_IBM_STAGING_HGP: staging hub/group/project to use.
-        * QISKIT_IBM_STAGING_DEVICE: staging device to use.
-        * QISKIT_IBM_STAGING_PRIVATE_HGP: staging hub/group/project to use for private jobs.
-"""
+"""Decorators used by unit tests."""
```

Comment:
In order to introduce support for running the integration tests against different APIs (legacy, Cloud) separately I consolidated the decorators we previously had.

*Notable changes*
- best described in code, see below

Let's discuss objections related to decorator changes in this thread.